src/components/BeaconsTable.tsx

```diff
@@ -0,0 +1,173 @@
+  LastPage,
+  Remove,
+  SaveAlt,
+  Search,
+  ViewColumn,
+} from "@material-ui/icons";
+import MaterialTable, { Icons, MTableBodyRow } from "material-table";
+import React, {
+  forwardRef,
+  FunctionComponent,
+  useEffect,
+  useState,
+} from "react";
+import { IBeacon } from "../entities/IBeacon";
+import { IBeaconsGateway } from "../gateways/IBeaconsGateway";
+import { formatDate } from "../use-cases/formatDate";
+import { formatUses } from "../use-cases/formatUses";
+import { titleCase } from "../utils";
+
+interface IBeaconsTableProps {
+  beaconsGateway: IBeaconsGateway;
+}
+
+interface IBeaconsTableState {
+  isLoading: Boolean;
+  error: Boolean;
+  beacons: IBeacon[];
+}
+
+export const BeaconsTable: FunctionComponent<IBeaconsTableProps> = ({
+  beaconsGateway,
+}): JSX.Element => {
+  const tableIcons: Icons = {
+    Add: forwardRef((props, ref) => <AddBox {...props} ref={ref} />),
+    Check: forwardRef((props, ref) => <Check {...props} ref={ref} />),
+    Clear: forwardRef((props, ref) => <Clear {...props} ref={ref} />),
+    Delete: forwardRef((props, ref) => <DeleteOutline {...props} ref={ref} />),
+    DetailPanel: forwardRef((props, ref) => (
+      <ChevronRight {...props} ref={ref} />
+    )),
+    Edit: forwardRef((props, ref) => <Edit {...props} ref={ref} />),
+    Export: forwardRef((props, ref) => <SaveAlt {...props} ref={ref} />),
+    Filter: forwardRef((props, ref) => <FilterList {...props} ref={ref} />),
+    FirstPage: forwardRef((props, ref) => <FirstPage {...props} ref={ref} />),
+    LastPage: forwardRef((props, ref) => <LastPage {...props} ref={ref} />),
+    NextPage: forwardRef((props, ref) => <ChevronRight {...props} ref={ref} />),
+    PreviousPage: forwardRef((props, ref) => (
+      <ChevronLeft {...props} ref={ref} />
+    )),
+    ResetSearch: forwardRef((props, ref) => <Clear {...props} ref={ref} />),
+    Search: forwardRef((props, ref) => <Search {...props} ref={ref} />),
+    SortArrow: forwardRef((props, ref) => (
+      <ArrowDownward {...props} ref={ref} />
+    )),
+    ThirdStateCheck: forwardRef((props, ref) => (
+      <Remove {...props} ref={ref} />
+    )),
+    ViewColumn: forwardRef((props, ref) => <ViewColumn {...props} ref={ref} />),
+  };
+
+  const [state, setState] = useState<IBeaconsTableState>({
+    isLoading: true,
+    error: false,
+    beacons: [],
+  });
+
+  useEffect((): void => {
+    const fetchBeacons = async () => {
+      setState({ ...state, isLoading: true });
+      try {
+        const beacons = await beaconsGateway.getAllBeacons();
+        setState({
+          ...state,
+          isLoading: false,
+          error: false,
+          beacons,
+        });
+      } catch (error) {
+        setState({
+          ...state,
+          isLoading: false,
+          error: error?.message,
+        });
+      }
+    };
+
+    fetchBeacons(); // TODO: What events should cause new beacons to be fetched?  Currently once on first render
+  }, []); // eslint-disable-line
```

Comment:
`fetchBeacons` inside`[useEffect`](https://reactjs.org/docs/hooks-effect.html) makes the API call via `beaconsGateway`.  The empty dependency array at line 100 means it's only called at first render.  

I left a todo for us to consider what events should prompt `fetchBeacons()` to be called, e.g. a "sync" or "refresh" button?  Considering the edge case of a beacon is registered then immediately activated ("Well now that's my beacon registered.  I wonder what happens if I press this big red button......")